src/graphics/image_factory.cpp

```
@@ -17,44 +17,50 @@ namespace canyon::graphics {
-            nlohmann::json json;
-            try {
-                ifile >> json;
-            } catch (std::exception&) {
-                return false;
-            }
+        std::ifstream ifile(detailsPath);
+        if (!ifile.is_open()) {
+            return false;
+        }
+
+        nlohmann::json json;
```

Clang (14.0.6)

[error] variable 'json' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)